crates/core/tedge/src/cli/certificate/renew.rs

```diff
@@ -0,0 +1,78 @@
+use super::error::CertError;
+use crate::command::Command;
+use camino::Utf8PathBuf;
+use certificate::KeyCertPair;
+use certificate::NewCertificateConfig;
+use certificate::PemCertificate;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+use std::path::Path;
+use tedge_utils::paths::set_permission;
+use tedge_utils::paths::validate_parent_dir_exists;
+
+pub struct RenewCertCmd {
+    /// The path where the device certificate will be stored
+    pub cert_path: Utf8PathBuf,
+}
+
+impl Command for RenewCertCmd {
+    fn description(&self) -> String {
+        "Renew a self-signed certificate for the device.".into()
+    }
+
+    fn execute(&self) -> anyhow::Result<()> {
+        let config = NewCertificateConfig::default();
+        self.renew_test_certificate(&config)?;
+        eprintln!("Certificate was successfully renewed, for un-interrupted service, the certificate has to be uploaded to the cloud");
+        Ok(())
+    }
+}
+
+impl RenewCertCmd {
+    fn renew_test_certificate(&self, config: &NewCertificateConfig) -> Result<(), CertError> {
+        validate_parent_dir_exists(&self.cert_path).map_err(CertError::CertPathError)?;
+
+        // get the ID from the certificate
+        let cert = KeyCertPair::new_selfsigned_certificate(config, &self.device_id()?)?;
+
+        let mut cert_file =
+            create_new_file(&self.cert_path, crate::BROKER_USER, crate::BROKER_GROUP)
+                .map_err(|err| err.cert_context(self.cert_path.clone()))?;
+
+        let cert_pem = cert.certificate_pem_string()?;
+        cert_file.write_all(cert_pem.as_bytes())?;
+        cert_file.sync_all()?;
+
+        // Prevent the certificate to be overwritten
+        set_permission(&cert_file, 0o444)?;
+
+        Ok(())
+    }
+
+    fn device_id(&self) -> Result<String, CertError> {
+        let pem = PemCertificate::from_pem_file(&self.cert_path).map_err(|err| match err {
+            certificate::CertificateError::IoError(from) => {
+                CertError::IoError(from).cert_context(self.cert_path.clone())
+            }
+            from => CertError::CertificateError(from),
+        })?;
+        let device_id = pem.subject_common_name()?;
+
+        Ok(device_id)
+    }
+}
+
+fn create_new_file(path: impl AsRef<Path>, user: &str, group: &str) -> Result<File, CertError> {
+    let file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .open(path.as_ref())?;
+
+    // Ignore errors - This was the behavior with the now deprecated user manager.
+    // - When `tedge cert create` is not run as root, a certificate is created but owned by the user running the command.
+    // - A better approach could be to remove this `chown` and run the command as mosquitto.
+    let _ = tedge_utils::file::change_user_and_group(path.as_ref(), user, group);
+
+    Ok(file)
+}
```

Comment:
The function `certificate::create::create_new_file()` must be made public rather than duplicated.